inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java

```diff
@@ -486,6 +497,17 @@ private void handleDirtyData(Object dirtyData, DirtyType dirtyType, Exception e)
             }
             throw ex;
         }
+
+        if (multipleSink) {
```

Comment:
It is suggest to send the metic after the dirty data sink because the dirty sink maybe failed.